spec/features/edit_dataset_spec.rb

```diff
@@ -21,7 +21,7 @@
   context 'editing published datasets from show page' do
     before(:each) do
       click_link 'Manage datasets'
-      edit_dataset(:published_dataset)
```

Comment:
this `edit_dataset` helper was dependent on order, which made it brittle. Fetching the link by its `href` instead.